sycl/source/device_selector.cpp

```diff
@@ -81,6 +131,9 @@ int default_selector::operator()(const device &dev) const {
 
   int Score = -1;
 
+  if (isDeviceInvalidForBe(dev))
```

Comment:
should we instead filter non-SYCL_BE devices in platform_impl::get_devices?

Reply:
Thats a good suggestion. I've moved the bypassing of OpenCL to get_platforms, it is actually cleaner now.